### DISCUSSION TITLE
Retrieving Additional Information Automatically

### DISCUSSION THREAD

**Discussion by beeblebroccoli** [Ulauncher/Ulauncher#893] [Ideas]
**This isn't an issue, this is a feature request/inquiry.**

One of the features that I find to be useful in Mac's Spotlight is the program's ability to display information from external sources (Such as wikipedia, or a dictionary) directly in the search window.

Relevant image here:
![Examples of Macs Spotlight](https://www.howtogeek.com/wp-content/uploads/2014/11/img_5457b2175773e.jpg)

I was wondering if it would be possible to implement this feature or a feature similar to it, as it is extremely useful and saves me from having to start up a browser in order to find out more about a subject. It's also a feature that I have yet to find in other similar programs (Synapse, Albert, etc.), so if possible it would definitely give you a leg-up on other similar launchers.

I'm fully aware that adding this feature could constitute a complete reworking of the schedule, but if possible it would be great if at least the possibility for this functionality to be developed by a third party was available. Such functionality would include, eventually, the ability to section off search results into categories as shown in Mac's Spotlight (Sections for web results, dictionary results, wikipedia results, etc.), and the ability to add a section in the search results where large(r) bodies of text or images can be displayed. This way there'd be room to display the first paragraph or so from, say, a stack overflow post, some google instant search results, a wolfram alpha calculation, or a wikipedia page.

If this was added, it would make this system insanely useful as a lot of searches could be completed quickly through the launcher without having to resort to a heavy browser.

**Comment by gornostal** [comment]
That's a great feature request. It's definitely possible to implement that, but after the plugin API.

**Comment by beeblebroccoli** [comment]
Alright, that's great! I've been searching for a usable launcher that has this functionality for a while. I haven't really found any that had everything I wanted, but if this were implemented I'd definitely switch to this as my primary launcher.

**Comment by unique1o1** [comment]
Hi, I just made a super simple extension for ulauncher for finding definition/ synonym for words.
[ulauncher-SearchDefination](https://github.com/unique1o1/ulauncher-SearchDefinition)

Hope it helps.

_Currently it doesn't support external sources like wikipedia_

**Comment by gornostal** [comment]
@unique1o1 looks great :+1:

**Comment by decoherencer** [comment]
I am guessing there is no extension-api to display the larger result item in ulauncher yet?

**Reply by friday** [reply]
No. This is not something you can do with an extension and it's not even on the roadmap. There's a lot of hoops to jump through to get this working on the API level, platform compatibility, including severe performance regressions for needing webkit2gtk in case we want to support free form html and css in the preview.

So realistically this will never happen, or will happen in a limited form in a long time from now.
